Improve combine_results script

ktbarrett opened the pull request:
Follow on from #3791.

* Allow multiple directories to be provided to script to search for results.xml files.
* Removed --suppress-rc, just use the shell to ignore the return code if desired.
* Typing and public API cleanup.
* Cleanup configuration of ArgumentParser.
* Added newsfragment.
* Added documentation.

ktbarrett commented on docs/source/newsfragments/3791.feature.rst:
```diff
@@ -0,0 +1 @@
+The ``bin/combine_results.py`` script from the repo now ships with the cocotb installation. See usage by calling ``python -m cocotb_tools.combine_results -h``.
```
Add a link to the documentation.

ktbarrett commented on src/cocotb_tools/combine_results.py:
```diff
@@ -1,110 +1,84 @@
+        "directories",
+        nargs="*",
+        default=["."],
+        help="Path to base directory to search from",
```
```suggestion
        help="Directories to search for 'result.xml' files",
```

marlonjames replied:
?
```suggestion
        help="Paths to one or more base directory to search from",
```

ktbarrett commented on src/cocotb_tools/combine_results.py:
```diff
@@ -1,110 +1,84 @@
+                        "package"
+                    ) == ts.get("package"):
+                        if args.verbose:
+                            print("Already found.")
```
What is this doing exactly?

ktbarrett replied:
Saying whether the testsuite (whose info was just printed) was already found in the results. In that case it extends that testsuite rather than adding it "new" to the results.

marlonjames replied:
If the `name` and `package` of a `testsuite` has already been found, it merges the `testcase` elements into the previously found one.

ktbarrett reviewed (commented):
Add an option to handle alternative naming of results file since we support `COCOTB_RESULTS_FILE`.

ktbarrett commented:
I added `--input-filenames` to allow the user to specify a filename other than `results.xml` since they can do that with `COCOTB_RESULTS_FILE`. I renamed `--testsuites-name` to `--output-testsuites-name` for clarity. And I added `-i` and `-o` short aliases for `--input-filenames` and `--output-file`.
